flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -601,8 +601,9 @@ class ScalarExprLowering {
       if (Fortran::semantics::IsPointer(sym)) {
         auto initialTarget = Fortran::lower::genInitialDataTarget(
             converter, loc, componentTy, expr.value());
-        res = builder.create<fir::InsertValueOp>(loc, recTy, res, initialTarget,
-                                                 field);
+        res = builder.create<fir::InsertValueOp>(
+            loc, recTy, res, initialTarget,
+            builder.getArrayAttr(field.getAttributes()));
```

Comment:
Is this the getter method (getAttributes) that @clementval talked about in slack?

Also, is it possible to remove the FieldIndexOp creation if it is only used to create the attributes?

Reply:
Yeah that's what I was talking about. If the FieldIndexOp is not necessary here and you have the attr you can remove it I guess.

Reply:
@schweitzpgi is it OK to have the FieldIndexOp for creating the attributes or is there a direct way (without using the FieldIndexOp) to create the attributes?

Reply:
This will need to be reworked in F2003. A field index can have runtime values as parameters and therefore cannot simply be converted to a compile-time constant. (See the TODO on line 596.)

It is ok for now to just use the attributes from the field here.